code/modules/vehicles/unmanned/unmanned_vehicle.dm

```diff
@@ -235,14 +245,24 @@
 		in_chamber.original_target = target
 		in_chamber.def_zone = pick("chest","chest","chest","head")
 		//Shoot at the thing
-		playsound(loc, gunnoise, 75, 1)
+		var/angle = Get_Angle(src, target)
+		playsound(loc, gunnoise, 65, 1)
 		in_chamber.fire_at(target, src, null, ammo.max_range, ammo.shell_speed)
 		in_chamber = null
 		COOLDOWN_START(src, fire_cooldown, fire_delay)
 		current_rounds--
+		flash.transform = null
+		flash.transform = turn(flash.transform, angle)
+		src.vis_contents += flash
+		//Removes muzzle flash
+		addtimer(CALLBACK(src, .proc/delete_muzzle_flash, src, flash), 0.3 SECONDS)
 		hud_set_uav_ammo()
 	return TRUE
 
+/obj/vehicle/unmanned/proc/delete_muzzle_flash(atom/movable/src,atom/movable/vis_obj/effect/muzzle_flash/flash)
```

Comment:
This is not how src is used like, at all
1) src is an innate var, why are you making it an arg
2) use spacing when doing args
3) you do not need any of these args at all because all the needed vars are on the vehicle